packages/exchange/tests/providers/test_google.py

```diff
@@ -138,3 +138,18 @@ def test_google_tools_integration():
     assert tool_use.id is not None
     assert tool_use.name == "read_file"
     assert tool_use.parameters == {"filename": "test.txt"}
+
+
+@pytest.mark.vcr()
+def test_google_vision(default_google_env):
+    reply_message, reply_usage = vision(GoogleProvider, GOOGLE_MODEL)
+
+    assert reply_message.content == [Text(text='The first entry in the menu says "Ask Goose 🦆". \n')]
+    assert reply_usage.total_tokens == 298
```

Comment:
Actually this looks suspicious as total_tokens seems like it should include the image, so the number would be higher. Maybe there's something to document or double-check vs openai here?

Reply:
hmm nevermind even if surprising, the parsing is correct, as the API isn't talking about the image attachment in terms of tokens!

```
\n  \"usageMetadata\": {\n    \"promptTokenCount\":
        285,\n    \"candidatesTokenCount\": 13,\n    \"totalTokenCount\": 298\n  }\n}\n"
```